module/documents/item.mjs

```diff
@@ -1165,10 +1167,12 @@ export default class Item5e extends SystemDocumentMixin(Item) {
 
     // Consume Spell Slots
     if ( config.consumeSpellSlot ) {
-      const level = this.actor?.system.spells[config.slotLevel];
+      const spellData = this.actor?.system.spells ?? {};
+      const level = spellData[config.slotLevel];
       const spells = Number(level?.value ?? 0);
       if ( spells === 0 ) {
-        const labelKey = config.slotLevel === "pact" ? "DND5E.SpellProgPact" : `DND5E.SpellLevel${this.system.level}`;
+        const isLeveled = /spell[0-9]+/.test(config.slotLevel || "");
```

Comment:
I know a lot of these regexes are from existing code but might as well try to clean up a bit as we go along:

```suggestion
        const isLeveled = /spell\d+/.test(config.slotLevel || "");
```